classes/Village.php

```diff
@@ -129,4 +135,287 @@ public function getRelations(System $system, string $village_id): array {
         }
         return $relations;
     }
+
+    /**
+     * @return array
+     */
+    public static function getVillagePopulation(System $system, string $village): array {
+        $villager_counts = [];
+        $villager_result = $system->db->query(
+            "SELECT
+                COUNT(IF(`rank`=1,1,NULL)) as `academy`,
+                COUNT(IF(`rank`=2,1,NULL)) as `genin`,
+                COUNT(IF(`rank`=3,1,NULL)) as `chuunin`,
+                COUNT(IF(`rank`=4,1,NULL)) as `jonin`
+                FROM `users` WHERE `village` = '{$village}'"
+        );
+		$villager_result = $system->db->fetch($villager_result);
+        foreach ($villager_result as $key => $value) {
+            $villager_counts[] = [
+                "rank" => $key,
+                "count" => (int)$value,
+            ];
+        }
+        return $villager_counts;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getVillagePopulationTotal(System $system, string $village): int
+    {
+        $villager_result = $system->db->query("SELECT COUNT(*) AS count FROM `users` WHERE `village` = '{$village}'");
+        $villager_result = $system->db->fetch($villager_result);
+        return $villager_result['count'];
+    }
+
+    /**
+     * @return bool
+     */
+    public static function claimSeat(System $system, User $player, string $seat_type): string {
```

Comment:
Could you extract this logic function to a `VillageManager` class so the `Village` class itself can be closer to a static data object?

I think other "write" functions like `updateResources` are good candidates as well

Reply:
That makes sense, `Village` class bloated very fast. 
Will move most of the functionality into `VillageManager`.